### PR TITLE
Layout switcher

### PR DESCRIPTION
**Pull request by mrpatpat**
# Layout Switcher

## Description

Add possibility to switch layouts in order to integrate other visualization methods. Issue #141 

Integrates angularjs ui router in order to switch between layouts. It does not use the routers url matching functions.

Adds CodeCharta and TestVille layouts to experimental tab. The correct implementation of the TestVille Layout will be done in another issue.

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

* [x] **All** requirements mentioned in the issue are implemented
* [x] Does match the Code of Conduct and the Contribution file 
* [x] Task has its own **GitHub issue** (something it is solving)
  * [x] Issue number is **included in the commit messages** for traceability
* [x] **Update the README.md** with any changes/additions made
* [x] **Update the CHANGELOG.md** with any changes/additions made
* [x] **Enough test coverage to ensure that uncovered changes do not break functionality**
* [x] **All tests pass**    
* [x] **Descriptive pull request text**, answering:
  + What problem/issue are you fixing?
  + What does this PR implement and how? 
* [x] **Assign your PR to someone** for a code review
  + This person _will be contacted **first**_ if a bug is introduced into `master`
* [x] **Manual testing** did not fail

### CODE REVIEW

File: visualization/app/codeCharta/codeCharta.component.scss [hunk 1]
```diff
@@ -1,6 +1,5 @@
 @import "../mixins";
 
-code-charta-component {
```
Comment (alschmut) [inline visualization/app/codeCharta/codeCharta.component.scss]: By removing this the sidebar doesn't adapt the with of 600px anymore...

Comment (mrpatpat) [reply visualization/app/codeCharta/codeCharta.component.scss]: I will fix this.

**Comment by mrpatpat** [comment]
We should disable the  view since it has no purpose yet

**Comment by Richargh** [comment]
Can we change the code so the new layout works even for regular cc.json? My understanding is that roads can be folders/packages. Then we won't have to hide it and can even provide it as an alternate layout.

**Pull request merged**